Template10 (Library)/Services/NavigationService/NavigationService.cs

```diff
@@ -290,8 +299,16 @@ public void Navigate<T>(T key, object parameter = null, NavigationTransitionInfo
             state.Write<string>("NavigateState", FrameFacadeInternal?.NavigationService.NavigationState);
         }
 
-        public event TypedEventHandler<Type> AfterRestoreSavedNavigation;
-        public async Task<bool> RestoreSavedNavigationAsync()
+        public event TypedEventHandler<Type> OnNavigationStateLoaded;
```

Comment:
Thanks for the review, Jerry. I've changed the event name back - when we start work on 2.x, I'll probably submit a PR refactoring NavigationService. Among those changes will be:

 - Methods based on the assignable implementation pattern
 - New events for saving/loading

By the way, I have discovered a problem with the assignable implementation pattern. Because of C#'s design, we can't create methods using generics via this pattern. Not only that, but it is difficult to add optional parameters.